### PR TITLE
Add r-survtmle

### PR DESCRIPTION
**Pull request by leej3**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by leej3** [comment]
@conda-forge/help-r

File: recipes/r-survtmle/LICENSE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+YEAR: 2016
```
Comment (mfansler) [inline recipes/r-survtmle/LICENSE:1]: This file can be removed - it will come with the tarball.

File: recipes/r-survtmle/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+# 'Autobrew' is being used by more and more packages these days
```
Comment (mfansler) [inline recipes/r-survtmle/build.sh:3]: Remove comments.

File: recipes/r-survtmle/build.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# to grab static libraries from Homebrew bottles. These bottles
+# are fetched via Homebrew's --force-bottle option which grabs
+# a bottle for the build machine which may not be macOS 10.9.
+# Also, we want to use conda packages (and shared libraries) for
+# these 'system' dependencies. See:
+# https://github.com/jeroen/autobrew/issues/3
+export DISABLE_AUTOBREW=1
+
+# R refuses to build packages that mark themselves as Priority: Recommended
+mv DESCRIPTION DESCRIPTION.old
```
Comment (mfansler) [inline recipes/r-survtmle/build.sh:13]: The `Priority` thing is not considered an issue anymore. Remove.

File: recipes/r-survtmle/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  url:
+    - {{ cran_mirror }}/src/contrib/survtmle_{{ version }}.tar.gz
+    - {{ cran_mirror }}/src/contrib/Archive/survtmle/survtmle_{{ version }}.tar.gz
+  sha256: d286799265f35f2ae7631f2f1714936ff2b1866cdd95a2848c252482f9ce23d0
+
+build:
+  merge_build_host: True  # [win]
+  # If this is a new build for the same version, increment the build number.
+  number: 0
+
```
Comment (mfansler) [inline recipes/r-survtmle/meta.yaml:20]: Should be a `noarch` build. Add

```yaml
  noarch: generic
```

File: recipes/r-survtmle/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+source:
+  url:
+    - {{ cran_mirror }}/src/contrib/survtmle_{{ version }}.tar.gz
+    - {{ cran_mirror }}/src/contrib/Archive/survtmle/survtmle_{{ version }}.tar.gz
+  sha256: d286799265f35f2ae7631f2f1714936ff2b1866cdd95a2848c252482f9ce23d0
+
+build:
+  merge_build_host: True  # [win]
+  # If this is a new build for the same version, increment the build number.
```
Comment (mfansler) [inline recipes/r-survtmle/meta.yaml:18]: Remove comments.

File: recipes/r-survtmle/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    without competing risks, including estimators that respect bounds (Benkeser, Carone,
+    and Gilbert. Statistics in Medicine, 2017. <doi:10.1002/sim.7337>).
+  license_family: MIT
+  license_file:
+    - '{{ environ["PREFIX"] }}/lib/R/share/licenses/MIT'
+    - LICENSE
+
+extra:
+  recipe-maintainers:
+    - leej3
```
Comment (mfansler) [inline recipes/r-survtmle/meta.yaml:73]: All R packages should include `conda-forge/r` as a maintainer.

File: recipes/r-survtmle/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  # This is required to make R link correctly on Linux.
+  rpaths:
+    - lib/R/lib/
+    - lib/
+
+# Suggests: testthat, knitr, rmarkdown, survival, cmprsk, tibble
+requirements:
+  build:
+    - {{ posix }}zip               # [win]
+
```
Comment (mfansler) [inline recipes/r-survtmle/meta.yaml:30]: Be prepared for cross-compiled builds by including:

```yaml
    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
```

**Review by mfansler — CHANGES_REQUESTED** [review]
Thanks for the submission. It appears an older recipe generator was used. All suggested changes would be fixed if the latest [`conda_r_skeleton_helper`](https://github.com/bgruening/conda_r_skeleton_helper) were used instead.

**Comment by leej3** [comment]
Thanks for the tips. I will use that skeleton helper in the future. Fixes applied...

Regarding `noarch: generic` I was unsure because there is byte-code compiled as part of the package creation. I couldn't find extensive docs to describe how one decides upon a packages noarch status. Are there any docs you could point me to on this? Similarly, I can't find any docs regarding support of cross-compilation...

**Review by mfansler — APPROVED** [review]

**Comment by mfansler** [comment]
[The DESCRIPTION file](https://github.com/cran/survtmle/blob/295de8b4be182bc6696afd2ef06f0f738d3a5eae/DESCRIPTION#L28) reports no compilation.

There are some sections in the Knowledge Base that pertain to targeting additional platforms, e.g., [**osx-arm64**](https://conda-forge.org/docs/maintainer/knowledge_base.html#apple-silicon-builds), [**linux-aarch64**](https://conda-forge.org/docs/maintainer/knowledge_base.html#using-arch-rebuild-txt). Admittedly, they don't describe in detail the cross-compilation strategy. For working purposes, new recipes in `staged-recipes` should cover just the default supported platforms. After merging and feedstock creation, the package can be added to the desired migration list and a migration bot will create a PR that converts the recipe to support the requested platform. So, it's mostly automated.

**Comment by mfansler** [comment]
@conda-forge/staged-recipes looks good to me

**Pull request merged**